server/app/controllers/contact-controller.js

```diff
@@ -0,0 +1,76 @@
+const Joi = require('joi');
+const mailService = require('../services/mail-service');
+const { createServerError, createUserError } = require('../../utils');
+const logger = require('../../logger');
+const request = require('request');
+const CONFIG = require('../../config/contactForm');
+
+/**
+ * Verify request and send form to the appropriate user(s).
+ *
+ * @param req Request
+ * @param res Response
+ * @return {Promise.<void>} Nothing
+ */
+async function sendForm(req, res) {
+    const schema = Joi.object().keys({
+        contactFormName: Joi.string().valid('concert', 'event', 'lost', 'website').required(),
+        values: Joi.object().required(),
+        token: Joi.string().required(),
+    });
+
+    const { error } = schema.validate(req.body);
+    if (error) throw createUserError('BadRequest', error.message);
+
+    var options = {
+        method: 'POST',
+        url: 'https://www.google.com/recaptcha/api/siteverify',
+        qs: {
+            secret: CONFIG.RECAPTCHA.SECRET,
+            response: req.body.token
+        }
+    };
+
+    const verif = await new Promise((resolve, reject) => {
+        request(options, (error, response, body) => {
+            if (error) {
+                reject();
+            }
+            resolve(JSON.parse(body));
+        });
+    }).catch(() => {
```

Comment:
I'm not sure if it is the best way to catch the callback error ? @Embraser01

Reply:
Actually you should not use the package `request` but the package `request-native-promise`. Then it will be a promise https://github.com/request/request-promise-native

Reply:
Ok :)